umh-core/pkg/fsm/dataflowcomponent/models.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2025 UMH Systems GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dataflowcomponent
+
+import (
+	"context"
+	"time"
+
+	internalfsm "github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/internal/fsm"
+	benthosserviceconfig "github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/config/benthosserviceconfig"
+	"github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/constants"
+	publicfsm "github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/fsm"
+)
+
+// Operational state constants (using internal_fsm compatible naming)
+const (
+	// OperationalStateStopped is the initial state and also the state when the service is stopped
+	OperationalStateStopped = "stopped"
+
+	// OperationalStateStarting is the state when the component is starting
+	OperationalStateStarting = "starting"
+
+	// OperationalStateActive is the state when the component is active and running
+	OperationalStateActive = "active"
+
+	// OperationalStateDegraded is the state when the component is running but has encountered issues
+	OperationalStateDegraded = "degraded"
+
+	// OperationalStateStopping is the state when the component is in the process of stopping
+	OperationalStateStopping = "stopping"
+)
+
+// Operational event constants
+const (
+	// Basic lifecycle events
+	EventStart     = "start"
+	EventStartDone = "start_done"
+	EventStop      = "stop"
+	EventStopDone  = "stop_done"
+
+	// Status events
+	EventDegraded  = "degraded"
+	EventRecovered = "recovered"
+)
+
+// IsOperationalState returns whether the given state is a valid operational state
+func IsOperationalState(state string) bool {
+	switch state {
+	case OperationalStateStopped,
+		OperationalStateStarting,
+		OperationalStateActive,
+		OperationalStateDegraded,
+		OperationalStateStopping:
+		return true
+	}
+	return false
+}
+
+// DataFlowComponentConfig represents the configuration for a data flow component
+type DataFlowComponentConfig struct {
+	// Basic component configuration
+	Name         string `yaml:"name"`
+	DesiredState string `yaml:"desiredState"`
+
+	// Service configuration similar to BenthosServiceConfig
+	ServiceConfig benthosserviceconfig.BenthosServiceConfig `yaml:"serviceConfig"`
+}
+
+// DFCObservedState contains the observed state of a DataFlowComponent
+type DFCObservedState struct {
+	// Component configuration in the benthos config
```

Comment:
shouldn't we expose all the underlying observed state as well? so that we can access the metrics, etc.?